[W-14231361] Configuring Grafana Dashboards (#350)
mulesoft/docs-private-cloud: modules/ROOT/pages/grafana-dashborads.adoc

> = Configuring Grafana Dashboards

Reviewer: The filename should probably be grafana-dashboards.adoc

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> Anypoint Platform Private Cloud Edition (Anypoint Platform PCE) includes Grafana dashboards, which provide a visual representation of the data collected by the monitoring tools in your Anypoint Platform PCE installation. These dashboards help you track the health of your Anypoint Platform PCE installation and its running applications

Reviewer: Can you flip this to put the user value first?

> Anypoint Platform Private Cloud Edition (Anypoint Platform PCE) includes Grafana dashboards, which provide a visual representation of the data collected by the monitoring tools in your Anypoint Platform PCE installation. These dashboards help you track the health of your Anypoint Platform PCE installation and its running applications

Reviewer: ```suggestion
Monitor the health of your Anypoint Platform Private Cloud Edition (Anypoint Platform PCE) installation and its running applications with Grafana dashboards. These dashboards provide a visual representation of the data collected by the monitoring tools in your Anypoint Platform PCE installation.
```

The Grafana application uses a Kubernetes service of type `NodePort` to expose the service to the external network. To access Grafana, configure the load balancer to forward traffic to port `32380`.

Alternatively, use the `kubectl port-forward` command to access Grafana locally:

[source,bash]
----
kubectl -n monitoring port-forward $(kubectl -n monitoring get svc --selector='app.kubernetes.io/name=grafana' -oname | head -n 1) 3000:80
----

== Access the Grafana Dashboard

> To retrieve credentials for Grafana, use the following commands to get the username and password:

Reviewer: "retrieve credentials" and "get the username and password" are the same thing.

> To retrieve credentials for Grafana, use the following commands to get the username and password:

Reviewer: ```suggestion
To retrieve credentials for Grafana, use these commands:
```

[source,bash]
----
kubectl -n monitoring get secrets grafana-creds -ojsonpath='{.data.admin-user}' | base64 --decode
kubectl -n monitoring get secrets grafana-creds -ojsonpath='{.data.admin-password}' | base64 --decode
----

== Create a Custom Dashboard

> You can create custom dashboards in Grafana using its web interface. However, these dashboards are lost if the Grafana pod restarts. To create a custom dashboard that persists across restarts, create a `ConfigMap` with the label `grafana_dashboard=1`.

Reviewer: GLOBAL: [Don't use labels as parts of speech](https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#:~:text=Do%20not%20use%20a%20UI%20label%20to%20replace%20normal%20sentence%20syntax%3A). 

create a `ConfigMap` element or file (or whatever it is)

This example demonstrates how to create a `ConfigMap` with a custom dashboard:

. Create a file with the dashboard configuration, for example, `example-redis.yaml`:

[source,yaml]
----
include::../assets/src/dashboard/example-redis.yaml[]
----

> [start=2]

Reviewer: Do you need this hard-coded numbering?

If you use + between the lines, the auto-numbering works...

Here's the info on [list continuation](https://confluence.internal.salesforce.com/display/MTDT/Formatting+with+Asciidoc#FormattingwithAsciidoc-ImplementListContinuation(akaIndentedParagraphs)).

> . Create the ConfigMap with the following command:

Reviewer: Here, you don't include the font change.

Is ConfigMap a proper noun? Or maybe it's a "configuration map"?

[source,bash]
----
kubectl apply -f example-redis.yaml
----

> To update the dashboard, modify the `ConfigMap`, and the dashboard is updated automatically.

Reviewer: ```suggestion
To update the dashboard, modify the `ConfigMap`, and the dashboard updates automatically.
```

Reviewer: Kim would say this is a use case for a semicolon.

== Delete a Custom Dashboard

To delete a custom dashboard, delete the corresponding `ConfigMap`. The dashboard associated with the `ConfigMap` is removed automatically.

[source,bash]
----
kubectl -n monitoring delete cm my-custom-dashboard
----

> == List All Persisted Dashboards

Reviewer: ```suggestion
== List All Persistent Dashboards
```

> To get a list of all ConfigMaps with the label `grafana_dashboard=1`, use the following command:

Reviewer: ```suggestion
To get a list of all ConfigMaps with the label `grafana_dashboard=1`, use this command:
```
